velox/expression/tests/ExpressionFuzzer.cpp

```diff
@@ -638,15 +680,40 @@ class ExpressionFuzzer {
       exceptionSimplifiedPtr = std::current_exception();
     }
 
-    // Compare results or exceptions (if any). Fail is anything is different.
-    if (exceptionCommonPtr || exceptionSimplifiedPtr) {
-      // Throws in case exceptions are not compatible. If they are compatible,
-      // return false to signal that the expression failed.
-      compareExceptions(exceptionCommonPtr, exceptionSimplifiedPtr);
-      return false;
-    } else {
-      // Throws in case output is different.
-      compareVectors(commonEvalResult.front(), simplifiedEvalResult.front());
+    try {
+      // Compare results or exceptions (if any). Fail is anything is different.
+      if (exceptionCommonPtr || exceptionSimplifiedPtr) {
+        // Throws in case exceptions are not compatible. If they are compatible,
+        // return false to signal that the expression failed.
+        compareExceptions(exceptionCommonPtr, exceptionSimplifiedPtr);
+        return false;
+      } else {
+        // Throws in case output is different.
+        compareVectors(commonEvalResult.front(), simplifiedEvalResult.front());
+      }
+    } catch (...) {
+      if (!FLAGS_repro_persist_path.empty()) {
+        std::string inputPath;
+        std::string resultPath;
+        std::string sqlPath;
+        exec::ExprSaver::saveData(
+            rowVector.get(),
+            FLAGS_repro_persist_path.c_str(),
+            inputPath,
+            inputPath);
+        if (copiedResult) {
```

Comment:
Why do we need to save the result vector? BTW, copying flattens the vector and we lose information about the encodings.

Reply:
Dirty initialized result vector sometimes can be the key to some failures, just like last time I fixed the bug greatest() has by not clearing result vector before writing into it.
Are there any deep copy that preserves everything?

Reply:
> Are there any deep copy that preserves everything?

Only serializing using the new VectorSaver.